frontend/public/components/cloud-services/catalog-source.tsx

```diff
@@ -21,28 +39,30 @@ export const PackageHeader: React.SFC<PackageHeaderProps> = (props) => <ListHead
 </ListHeader>;
 
 export const PackageRow: React.SFC<PackageRowProps> = (props) => {
-  const {obj, catalogSource, currentCSV, namespace, allPkgSubscriptions = []} = props;
+  const {obj, catalogSource, currentCSV, subscription} = props;
   const {displayName, icon = [], version} = currentCSV.spec;
-
+  const ns = getActiveNamespace();
   const channel = !_.isEmpty(obj.defaultChannel) ? obj.channels.find(ch => ch.name === obj.defaultChannel) : obj.channels[0];
-  const subscriptions = allPkgSubscriptions.filter(sub => sub.metadata.namespace === namespace || _.isEmpty(namespace));
+
+  const subscriptionLink = () => ns
+    ? <Link to={`/k8s/ns/${ns}/${SubscriptionModel.plural}/${subscription.metadata.name}`}>View subscription</Link>
+    : <Link to={`/k8s/all-namespaces/${SubscriptionModel.plural}?name=${obj.packageName}`}>View subscription(s)</Link>;
```

Comment:
Consider `resourcePath` to help build these URLs

Reply:
I don't like `resourcePath` right now; it fails if called before API discovery finishes registering the models. We can fix that in another PR.

Reply:
Yeah, understood. Generally, though, I'd like to keep the resource URL path logic in one place. It's less error prone. We can fix it later.